.yarn/patches/@polkadot-types-npm-10.5.1-87f9037bb0.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/types/extrinsic.d.ts b/types/extrinsic.d.ts
+index 0ad479fe3dd04f2fbd8704fdeeab8dcfdf2daafd..c9ee33c6465911394eea413317dc7b1c1ac326b0 100644
+--- a/types/extrinsic.d.ts
++++ b/types/extrinsic.d.ts
+@@ -101,7 +101,7 @@ export interface SignerResult {
+     /**
+      * @description The id for this request
+      */
+-    id: number;
++    id: string;
```

Comment:
Hmm you're right - I missed the fact that `Signer` is indirectly exposed to the dapps. So number it is. And we have to store it in the async local storage, so we've got to cope with the race conditions of reading and incrementing the id. Does such a function look fine to you?
```ts
const getNextId = (() => {
  let nextIdPromise = Promise.resolve(0); // "0" is just a placeholder - it'll never be returned from the function

  return () => {
    nextIdPromise = nextIdPromise.then(() => localStorageStores.signerRequestIds.update((lastId) => lastId + 1));

    return nextIdPromise;
  };
})();
```
This is supposed to be getting the ids serially, no matter _how parallel_ the invocations are. I'm assuming DDOSing the storage with requests is not a case, so the _pipe_ shouldn't get clogged.

Reply:
That's a really nice idea! perhaps it could be a module that exports the getter? not sure, but I find IIFE harder to read:
```
let nextIdPromise = Promise.resolve(0);

export const getNextId = () => {
    nextIdPromise = nextIdPromise.then(() => localStorageStores.signerRequestIds.update((lastId) => lastId + 1));

    return nextIdPromise;
};
```

Reply:
I'd usually agree, but in this case it's such a small util in such a short module, on top of that located in a very generic catalog, that I think it's not worth it. So I'd leave it if you don't mind.

Reply:
Actually, I might have over-engineered this particular spot with `getNextId`. After deeper analysis the module-global variable which `getNextId` was trying to eliminate is located in the web page, not the extension context, so it does is not affected by the manifest migration. That solution is not super-robust either, because it means that the _counter_ is resetted upon a refresh, but since it was like that already - I'm gona leave it like that and if we ever need to improve it, let it not be a part of this PR.